ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -496,6 +496,7 @@ def _sort_key(t, expr):
     ops.FloorDivide: _floor_divide,
     # other
     ops.SortKey: _sort_key,
+    ops.Date: unary(lambda arg: sa.cast(arg, sa.DATE)),
```

Comment:
Shouldn't we cast only if `arg` is not a `Date` already?
I'm not sure how `sa` handles it internally.

Reply:
Or ibis should prevent that scenario entirely?

Reply:
In this scenario, it's likely the database itself will do no additional work to cast if the input argument isn't a `DATE`.

In the long term I don't think ibis should do that. It's not clear to me that that's a safe operation in every case, and that it's worth any potential trouble.